tests/versions/tslib.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Validate the tslib and tsc version is consistent and corresponds to each other.
+ */
+import { execSync } from "child_process";
+import { readFileSync } from "fs";
+import JSON from "json5";
+import { join } from "path";
+
+type PackageInfo = {
+  name: string;
+  version: string;
+  private: boolean;
+  location: string;
+};
+
+console.log("Validating the tslib and tsc version is consistent within workspace.");
+const expectedVersions: { [name: string]: string } = JSON.parse(readFileSync(join(__dirname, "versions.json"), "utf8"));
+const packagesInfo: PackageInfo[] = JSON.parse(execSync("./node_modules/.bin/lerna list -l --json").toString());
```

Comment:
Instead of calling lerna with execSync, should we call subdirectories of workspaces listed in root package.json?

That way, we won't have a dependency on lerna from the test code.